src/client/pythonEnvironments/discovery/locators/services/windowsStoreLocator.ts

```diff
@@ -87,26 +88,27 @@ export async function isWindowsStoreEnvironment(interpreterPath: string): Promis
     return false;
 }
 
+/**
+ * This is a glob pattern which matches following file names:
+ * python3.8.exe
+ * python3.9.exe
+ * This pattern does not match:
+ * python.exe
+ * python2.7.exe
+ * python3.exe
+ * python38.exe
+ * 'python.exe', 'python3.exe', and 'python3.8.exe' can point to the same executable, hence only capture python3.*.exes.
+ */
+const pythonExeGlob = 'python3\.[0-9]*\.exe';
+
 /**
  * Checks if a given path ends with python3.*.exe. Not all python executables are matched as
  * we do not want to return duplicate executables.
  * @param {string} interpreterPath : Path to python interpreter.
  * @returns {boolean} : Returns true if the path matches pattern for windows python executable.
  */
-export function isWindowsStorePythonExe(interpreterPath:string): boolean {
-    /**
-     * This Reg-ex matches following file names:
-     * python3.8.exe
-     * python3.9.exe
-     * This Reg-ex does not match:
-     * python.exe
-     * python2.7.exe
-     * python3.exe
-     * python38.exe
-     */
-    const windowsPythonExes = /^python(3(.\d+))\.exe$/;
-
-    return windowsPythonExes.test(path.basename(interpreterPath));
+export function isWindowsStorePythonExe(interpreterPath: string): boolean {
+    return minimatch(path.basename(interpreterPath), pythonExeGlob, { nocase: true });
 }
```

Comment:
Why aren't we just using `isWindowsPythonExe()`?

Reply:
Because of the reason mentioned in the sprint demo today: We're matching only `python3.*.exes` so that we do collect only unique executables. `isWindowsPythonExe()` also matches with `python.exe` which is not what we want.

Reply:
The doc comment explains this.

Reply:
Then I'd expect the following to take care of it:

```suggestion
export function isWindowsStorePythonExe(interpreterPath: string): boolean {
    if (!isWindowsPythonExe(interpreterPath)) {
        return false;
    }
    return path.basename(interpreterPath) !== 'python.exe';
}
```

That way we don't accept executables we shouldn't, like "python3-whoa!!!.exe".

Reply:
Right, there's also `python3.exe`, and `python2.exe` that we would need to ignore. I can either do this or correct the glob pattern to not accept the whoaaa string.

Reply:
Will change the glob pattern to `python3\.[0-9]*([0-9])\.exe`.

Reply:
I can use `python3\.[0-9]*([0-9])\.exe`.

Reply:
Ah, so we're okay with "extended glob".  What you have looks fine, but I'd change the `*` to `?`.